token-erc-20/chaincode-go/chaincode/token_contract.go

```diff
@@ -84,11 +83,104 @@ func (s *SmartContract) Mint(ctx contractapi.TransactionContextInterface, amount
 	totalSupply += amount
 	ctx.GetStub().PutState(totalSupplyKey, []byte(strconv.Itoa(totalSupply)))
 
+	// Emit the Transfer event
+	transferEvent := map[string]string{"from": "0x0", "to": minter, "value": strconv.Itoa(amount)}
+	transferEventJSON, err := json.Marshal(transferEvent)
+	if err != nil {
+		return fmt.Errorf("failed to obtain JSON encoding: %v", err)
+	}
+	err = ctx.GetStub().SetEvent("Transfer", transferEventJSON)
+	if err != nil {
+		return fmt.Errorf("event failed to register: %v", err)
+	}
+
+	log.Printf("minter account %s balance updated from %d to %d", minter, currentBalance, updatedBalance)
+
 	return nil
 }
 
-// Transfer transfers tokens from client account to recipient account.
-// recipient account must be a valid clientID as returned by the ClientID() function.
+// Burn redeems tokens the minter's account balance
+// This function triggers a Transfer event
+func (s *SmartContract) Burn(ctx contractapi.TransactionContextInterface, amount int) error {
+
+	// Check minter authorization - this sample assumes Org1 is the central banker with privilege to burn new tokens
+	clientMSPID, err := ctx.GetClientIdentity().GetMSPID()
+	if err != nil {
+		return fmt.Errorf("failed to get MSPID: %v", err)
+	}
+	if clientMSPID != "Org1MSP" {
+		return fmt.Errorf("client is not authorized to mint new tokens")
+	}
+
+	// Get ID of submitting client identity
+	minter, err := ctx.GetClientIdentity().GetID()
+	if err != nil {
+		return fmt.Errorf("failed to get client id: %v", err)
+	}
+
+	if amount <= 0 {
+		return fmt.Errorf("burn amount must be a positive integer")
```

Comment:
Nit since there isn't any formatting.
```suggestion
		return errors.New("burn amount must be a positive integer")
```